src/main/java/com/github/wovnio/wovnjava/HtmlConverter.java

```diff
@@ -168,9 +162,9 @@ private void appendHrefLang() {
     }
 
     private void replaceNodeToMarkerComment(Element element) {
-        Comment comment = new Comment(WOVN_MARKER_PREFIX + String.valueOf(markers.size()));
-        element.replaceWith(comment);
-        markers.add(restore(element.outerHtml())); // restore original text if element has marker
+        String commentKey = htmlReplaceMarker.addCommentValue(htmlReplaceMarker.revert(element.html()));
+        element.html("");
```

Comment:
are you sure you can't simply do `element.html('<!-- foo -->')`?

Reply:
Nope, tried that. Raw `<!-- foo -->` isn't liked by jsoup, the `<` and `>` get's HTML encoded.